### DISCUSSION TITLE
Cannot bind super+period to `text:` actions

### DISCUSSION THREAD

**Discussion by scnewma** [ghostty-org/ghostty#2881] [Q&A] [answered]
## Summary

When attempting to bind a text output to the keybind `super+period`, I couldn't get any `text:` actions to work. Maybe I'm doing something wrong?

Example keybind:

```
keybind = clear
keybind = super+period=text:hello
```

## Expected Behavior

Pressing `super+.` outputs text "hello".

## Actual Behavior

Nothing happens, key input looks to be consumed (confirmed typing keybind in `xxd -psg`).

## Additional Information

I _can_ bind `super+period` to other actions, such as `new_tab`, and they work as expected.

```
keybind = clear
keybind = super+period=new_tab
```

I can also bind other keys to `text:hello` and they work as expected.

```
keybind = clear
keybind = super+l=text:hello
```

**Comment by mitchellh** [comment]
Thanks for the detailed information. I'm going to convert to a Q&A until this can be determined this is a bug.

**Reply by qinyuhang** [reply]
Hi, I opened the 'Terminal Inspect' and find out that ghostty is not out put correctly the super+period. Here is the screen recording. During the recording I use keycastr to display keypresses, it shows that 'cmd+.' is presses, but ghostty Terminal Inspect-Keyboard only output 'Press: Super+left_super'

https://github.com/user-attachments/assets/b6a174c0-6efb-4765-b8e0-886c04306af2

**Comment by abzrg** [comment]
I also cannot bind `super+period` to `esc:.` (on mac). I tried following combinations:

```
keybind = super+period=esc:period
keybind = super+period=esc:.
```

**Comment by denolfe** [comment]
I ran into [something similar](https://github.com/ghostty-org/ghostty/discussions/3858) and someone suggested it might be a duplicate of this one.

Interestingly, I was able to get `cmd+.` to bind for some actions, but specifically could not get it to work for `esc:.`. I thought I would add my details here.

To summarize:

- Verified binding cmd+. to an action works
- Verified binding the escape code to something other than cmd+. works.
- When combining cmd+. with esc:., no action occurs.

```
# WORKS: Binding  cmd+. to an action
keybind = cmd+.=toggle_quick_terminal

# WORKS: Binding cmd+b to esc:.
keybind = cmd+b=esc:.

# DOES NOT WORK: Binding cmd+. to esc:.
keybind = cmd+.=esc:.
```

**Comment by frantic1048** [comment]
**Improved Version:**

I have a similar issue with `cmd+period`, as I cannot unbind it (which prevents ghostty from consuming the `cmd+period` key combination).

What I'm doing is unbinding all keybindings and specifying a few keybinds. I am using a terminal multiplexer (zellij) and want to avoid keybinding conflicts between the terminal multiplexer and ghostty.

My config is as follows:

```
keybind = clear
keybind = cmd+v=paste_from_clipboard
keybind = cmd+d=close_window
keybind = cmd+q=quit
```

Output of `ghostty +list-keybinds`:

```
super + d  close_window
super + q  quit
super + v  paste_from_clipboard
```

**Comment by helloadj** [comment]
I have a similar issue with mac the key binding for ctrl+left  and ctrl+right does not work. 
my config looks like this.

keybind = clear
keybind = ctrl+right=esc:f
keybind = ctrl+left=esc:b

But binding to other key like the following works without any issue

keybind = shift+right=esc:f
keybind = shift+left=esc:b

Macbook M3 chip 
Macos: Sequoia 15.2

**Comment by mitchellh** [comment, marked as answer]
Reproduced #4540
